impl/src/testing.rs

```diff
@@ -215,5 +266,9 @@ pub fn mock_crate_index(crates: &HashMap<String, String>) -> TempDir {
     .unwrap();
   }
 
-  index_url_mock_dir
+  if mock_dir.is_none() {
```

Comment:
It seems this function always returns Some(x). Update it to not return Option anymore?

Reply:
maybe 

```rust
mock_dir.or_else(|| Some(index_url_mock_dir))
```

Reply:
No, it does return `None` some in cases
```
test planning::tests::test_binary_dependencies_vendored_genmode ... FAILED
test planning::tests::test_binary_dependencies_remote_genmode ... FAILED
```
I added a panic on that line and these tests tripped it.